contracts/IbbtcVaultZap.sol

```diff
@@ -197,11 +211,44 @@ contract IbbtcVaultZap is PausableUpgradeable {
         return _vaultShares(crvLp);
     }
 
+    function expectedAmount(uint256[4] calldata _amounts, bool _mintIbbtc)
+        public
+        view
+        returns (uint256 amount)
+    {
+        uint256[4] memory depositAmounts = _constructDeposit(
+            _amounts,
+            _mintIbbtc
+        );
+
+        if (depositAmounts[0] != 0) {
+            depositAmounts[0] = WIBBTC.sharesToBalance(depositAmounts[0]);
+        }
+
+        ERC20Upgradeable[4] memory assets = [
+            ERC20Upgradeable(address(IBBTC)),
+            ERC20Upgradeable(address(RENBTC)),
+            ERC20Upgradeable(address(WBTC)),
+            ERC20Upgradeable(address(SBTC))
+        ];
+        uint256 virtualPrice = CURVE_IBBTC_METAPOOL.get_virtual_price();
+        for (uint256 i = 0; i < 4; i++) {
+            if (depositAmounts[i] != 0) {
+                amount = amount.add(
+                    depositAmounts[i].mul(1e18).div(
+                        10**uint256(assets[i].decimals())
+                    )
+                );
+            }
+        }
+        amount = amount.mul(1e8).div(virtualPrice);
```

Comment:
Shouldn't this be `1e18`? Can also just multiply by `1e36` inside the loop (or multiply by `10 ** (36 - decimals)`)